modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -526,39 +524,51 @@ public void start() {
         });
     }
 
-    private void performRebalanceOnRecovery(long recoveryRevision) {
-        CompletableFuture<Void> pendingAssignmentsRecoveryFuture;
+    private void processAssignmentsOnRecovery(long recoveryRevision) {
+        var pendingAssignmentsPrefix = new ByteArray(PENDING_ASSIGNMENTS_PREFIX);
 
-        var prefix = new ByteArray(PENDING_ASSIGNMENTS_PREFIX);
+        startVv.update(recoveryRevision, (v, e) -> handleAssignmentsOnRecovery(
+                pendingAssignmentsPrefix,
+                recoveryRevision,
+                (entry, rev) -> handleChangePendingAssignmentEvent(entry, rev, true),
+                "pending"
+        ));
+    }
 
-        try (Cursor<Entry> cursor = metaStorageMgr.prefixLocally(prefix, recoveryRevision)) {
+    private CompletableFuture<Void> handleAssignmentsOnRecovery(
```

Comment:
As mentioned above `processAssignmentsOnRecovery` is only used to handle pending assignments and not the stable one. Thus either approach or naming is misleading.

Reply:
I think it will be reasonable after stable assignments will be processed on recovery as well, so I decided to leave it as is

Reply:
Ok